Add page for typescript client generator (#764)
api-platform/docs: client-generator/typescript.md

> # Typescript interfaces

Reviewer: ```suggestion
# Typescript Interfaces
```

# Typescript interfaces

> The Typescript Generator allows you to create typescript interfaces that you can embed in any Typescript-enabled project (Angular, Vue, etc...)

Reviewer: ```suggestion
The TypeScript Generator allows you to create [TypeScript interfaces](https://www.typescriptlang.org/docs/handbook/interfaces.html) that you can embed in any TypeScript-enabled project (React, Vue.js, Angular..)
```

To do so, first install the client generator.

You can do it locally...

    $ npm install @api-platform/client-generator

...or globally

    $ npm install -g @api-platform/client-generator

(If you are using `yarn`, use `yarn add` or `yarn global add`)

Then, in the app directory, generate the files for the resource you want:

>     $ generate-api-platform-client -g typescript https://demo.api-platform.com src/ --resource foo

Reviewer: You could use `npx` directly.

Author: Should I write it as an alternative way to invoke the command or should we replace the whole installation part in favor of `npx`?

Reviewer: I would replace it.

Author: Ok, I'll update it. 

By the way, I've tried to run the command through `npx` on my machine but it doesn't work.
```
luca@luca-mipro:~/Work/Personal/api-platform$ npx @api-platform/client-generator
npx: installed 284 in 10.132s
Cannot find module 'lodash'
```
However, the globally installed version works.

I'll open an issue in the client-generator repo to see if this is an issue that affects others or just me.

    # Replace the URL by the entrypoint of your Hydra-enabled API
    # "src/" represents where the interfaces will be generated
    # Omit the resource flag to generate files for all resource types exposed by the API

> This command parses the Hydra documentation and creates one `.ts` file for each Api Resource you have defined in your application, in the `interfaces` subfolder.

Reviewer: ```suggestion
This command parses the Hydra documentation and creates one `.ts` file for each API Resource you have defined in your application, in the `interfaces` subfolder.
```
